ontology/docs/faq.md

```diff
@@ -0,0 +1,115 @@
+
+Check to make sure the field to be defined isn’t already defined using some sub- or super-set of subfields. Note that each field is constructed with a grammar (for example, that point type is always the last subfield in the field), and it must be followed; see [here](ontology.md#fields) for further information. When deciding how to construct the field (that is, choosing how many and which subfields to use), there are some common situations covered [here](model_hvac.md#fields); keep in mind that it is best to be consistent, and so if a field already exists that follows a structure that can be reused, use it (for example, say you have a temperature sensor in a fireplace that you want to model: fields like `exhaust_air_temperature_sensor` exist, so reusing this form and replacing `exhaust` with `fireplace` will result in a consistent extension of the ontology).
+
+## How do you extend abstract types?
+**Construct the new abstract type by building it up from individual field combinations.** 
+
+When new functionality is required, it should be defined as an abstract type. The type should tightly model the specific functionality it is intended for; do not be overbroad, and do not try to represent multiple distinct functionalities -- use multiple inheritance for that.
+
+## How do you extend non-abstract types?
+**Construct the type by building it up from abstract function.** 
+
+If new subfields, fields, or abstract types are needed in order to be able to construct the type, then do so first. Types should try to be constructed such that all fields are applied via abstract types (that is, that no fields need to be defined directly on the type) but this is sometimes problematic; if a field is wanted, but does not merit being assigned to an abstract type, then it is acceptable to add it directly to the type (either as optional or required). For information on what is needed for a new type, see [here](ontology.md#entity-types).
+
+## What is the process for accepting extensions to the ontology?
+**If the extension passes the type validator and passes review by the team, it will be accepted.** 
+
+The type validator checks that additions or changes to the ontology do not break existing rules or compatibility (for example, the type validator will raise errors if a new field is added which is composed of undefined subfields). A review of the proposed extensions will be performed by the Digital Buildings team, which include building systems subject matter experts and software developers. The review covers both the subject matter (e.g. is a subfield too broadly defined for the intended application) and its holistic effect on the ontology (e.g. does it set precedent that makes further extension difficult). These reviews may require some deliberation and can result in proposed changes to the pull request, so it is better to submit pull requests in phases. See the next question for what is advised on ordering of ontology updates.
```

Comment:
cool I see it here. Should we add a section here on how this works? from here: http://go/digitalbuildings-sdk

Reply:
link doesnt work for me, not on my google machine... assume this is an internal doc?